src/zarr/core/metadata/__init__.py

```diff
@@ -1,4 +1,16 @@
-from .v2 import ArrayV2Metadata
-from .v3 import ArrayV3Metadata
+from typing import TypeAlias
 
-__all__ = ["ArrayV2Metadata", "ArrayV3Metadata"]
+from .v2 import ArrayV2Metadata, ArrayV2MetadataDict
+from .v3 import ArrayV3Metadata, ArrayV3MetadataDict
+
+ArrayMetadata: TypeAlias = ArrayV2Metadata | ArrayV3Metadata
```

Comment:
I know I suggested keeping `ArrayMetadata` as an alias for the union. I wonder now if it should be `ArrayMetadataT` to make it clearer that it's a type that can't be instantiated.

Likewise for `ArrayMetadataDictT`?

Reply:
done in https://github.com/zarr-developers/zarr-python/pull/2301/commits/1b612a88300908685d511e7105b6bf9dfad02526